doc/devdocs/types.rst

```diff
@@ -0,0 +1,522 @@
+****************
+More about types
+****************
+
+.. currentmodule:: Base
+
+If you've used Julia for a while, you understand the fundamental role
+that types play.  Here we try to get under the hood, focusing
+particularly on :ref:`parametric types <man-parametric-types>`.
+
+Types and sets (and ``Any`` and ``Union()``/``Bottom``)
+-------------------------------------------------------
+
+It's perhaps easiest to conceive of Julia's type system in terms of
+sets.  A concerete type corresponds to a single entity in the space of
+all possible types; an abstract type refers to a collection (set) of
+concrete types.  ``Any`` is a type that describes the entire universe
+of possible types; ``Integer`` is a subset of ``Any`` that includes
+``Int``, ``Int8``, and other concrete types.  Internally, julia also
+makes heavy use of another type known as ``Bottom``, or equivalently,
+``Union()``.  This corresponds to the empty set.
+
+Julia's types support the standard operations of set theory: you can
+ask whether ``T1`` is a "subset" (subtype) of ``T2`` with ``T1 <:
+T2``.  Likewise, you intersect two types using ``typeintersect`` and
+take their union with ``typejoin``::
```

Comment:
`Union` takes the union. `typejoin` usually produces a bigger type that is easier to represent.